optimize/backend/src/it/java/io/camunda/optimize/service/db/es/report/process/single/flownode/frequency/groupby/flownode/FlowNodeFrequencyByFlowNodeReportEvaluationIT.java.out

```diff
@@ -105,6 +105,86 @@
 //   }
 //
 //   @Test
+//   public void testCollapsedNodesAreExcludedWhenUnsavedReportViewedViewedAsHeatmap()
+//       throws IOException {
+//     // given
+//     String bpmnXml =
+//         BpmnModelUtil.getResourceFileAsString("/test/data/visible_and_collapsed_subprocess.bpmn");
+//     ProcessInstanceEngineDto processInstanceEngineDto =
+//         engineIntegrationExtension.deployAndStartProcess(BpmnModelUtil.parseBpmnModel(bpmnXml));
+//     importAllEngineEntitiesFromScratch();
+//
+//     // when
+//     final ProcessReportDataDto reportData =
+//         TemplatedProcessReportDataBuilder.createReportData()
+//             .setProcessDefinitionKey(processInstanceEngineDto.getProcessDefinitionKey())
+//             .setProcessDefinitionVersions(
+//                 List.of(processInstanceEngineDto.getProcessDefinitionVersion()))
+//             .setReportDataType(ProcessReportDataType.FLOW_NODE_FREQ_GROUP_BY_FLOW_NODE)
+//             .build();
+//     reportData.getConfiguration().setXml(bpmnXml);
+//     reportData.setVisualization(ProcessVisualization.HEAT);
+//     final ReportResultResponseDto<List<MapResultEntryDto>> result =
+//         reportClient.evaluateMapReport(reportData).getResult();
+//
+//     // then
+//     final List<MapResultEntryDto> resultData = result.getFirstMeasureData();
+//     // The flow nodes from the collapsed subprocess are not included
+//     assertThat(resultData).hasSize(7);
+//     final List<String> resultLabels = resultData.stream().map(MapResultEntryDto::getLabel).toList();
+//     assertThat(resultLabels)
```

Comment:
Q: Are there any result metrics that could theoretically be influenced by the flownode count? Eg relative frequency, things like that where we might use the flownode count for calculations. This is not affected by the filtering correct?

Reply:
I actually quite deliberately did not want to modify anything other than the view here. My reasoning is that the user could quite easily switch to another visualisation, such as a bar chart, and I don't think that any data itself should actually be different in this case. To me it would be odd behaviour for the user to change a visualisation and get a different result

Reply:
Yeah I agree I would also not want to have the metrics affected, I think my line of thinking was more to evaluate whether it makes sense to assert this in the test as well. The way I currently understand it though the filtering of the flownodes in the results is logically far enough removed from the calculation of the other metrics that an assertion isnt necessary